### PR TITLE
V0.16.0

### PR DESCRIPTION
**Pull request by ThummeTo**

### CODE REVIEW

**Comment by ThummeTo** [comment]
I will push a modified version here again @manuelbb-upb and @JoKircher : 
- I deleted some code and reused the similarity between the SimpleTypes and ScalarVariables (they are sharing the same argument sub-set called fmi2RealAttributes)
- there where some parsing functions that did almost the same as existing parsing functions, I replaced them
- I did some renaming to better fit the existing code
- I (temporary) removed the tests, because I removed the redundant structs
- ...

**Comment by ThummeTo** [comment]
This is WIP (tests are failing)

File: src/FMI2/ctype.jl [hunk 1]
```diff
@@ -84,15 +84,16 @@ export fmi2VariableDependency
 # Custom helper, not part of the FMI-Spec. 
 fmi2Unknown = fmi2VariableDependency
 
-abstract type fmi2ModelDescriptionVariable end
+# custom abstract types, not part of the FMI-spec
+abstract type fmi2Attributes end # extended version of the Attributes used in fmi2XXXAttributes
+abstract type fmi2AttributesExt end # default version of the Attributes used in fmi2XXXAttributes
```
Comment (JoKircher) [inline src/FMI2/ctype.jl:89]: What dos the `Ext` stand for? I would assume extended but that is what the comment behind `fmi2Attributes` indicates.

**Review by JoKircher — APPROVED** [review]
Thank you for this PR. I looks really clean and complient with the standard and the rest of this package

**Pull request merged**